src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/rc_helpers_test.cc

```diff
@@ -0,0 +1,278 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE. */
+
+#include "rc_rpc_plugin/rc_helpers.h"
+#include <map>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <vector>
+#include "gtest/gtest.h"
+#include "rc_rpc_plugin/rc_module_constants.h"
+
+namespace {
+const std::string kKeyId = "id";
+const std::string kKeyStatus = "status";
+const std::string kKeyDensity = "density";
+const std::string kKeyValue = "value";
+const std::string kArray = "array";
+}  // namespace
+
+namespace rc_rpc_plugin_test {
+using namespace rc_rpc_plugin;
+
+class RCHelpersTest : public ::testing::Test {
+ public:
+  RCHelpersTest() {}
+
+  smart_objects::SmartObject genSmartObject(const int& object_number,
+                                            const bool& use_id_key) {
+    smart_objects::SmartObject result(smart_objects::SmartType_Map);
+    result[kKeyStatus] = kKeyStatus + std::to_string(object_number);
+    result[kKeyValue] = kKeyValue + std::to_string(object_number);
+    if (use_id_key) {
+      result[kKeyId] = kKeyId + std::to_string(object_number);
+    }
+    return result;
+  }
+};
+
+TEST_F(RCHelpersTest, MergeModuleData_OneOfParamsNotSmartMap) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Array);
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  EXPECT_NE(result, data1);
+  EXPECT_EQ(result, data2);
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_FirstObjectDontContainKey) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Map);
+
+  std::map<std::string, std::string> expected_keys_values = {
+      {"key1", "value1"}, {"key2", "value2"}};
+  auto it = expected_keys_values.begin();
+
+  data1[it->first] = it->second;
+  ++it;
+  data2[it->first] = it->second;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  for (auto& e : expected_keys_values) {
+    EXPECT_TRUE(result.keyExists(e.first));
+    EXPECT_EQ(result[e.first], expected_keys_values[e.first]);
+  }
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_SecondObjectContainsDifferentValueType) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Map);
+
+  data1["key1"] = "value1";
+  data2["key1"] = 30;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  EXPECT_EQ(data2["key1"], result["key1"]);
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_Recursive) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Map);
+
+  data1["key1"]["subkey1"] = "subvalue1";
+  data2["key1"]["subkey2"] = 30;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
```

Comment:
@YarikMamykin const?

Reply:
@AByzhynar , fixed in b61ba7c099f2c1b7caf78e8180514ad3e9cf253a